tensorboard/webapp/widgets/card_fob/card_fob_controller_component.ts

```diff
@@ -252,27 +296,43 @@ export class CardFobControllerComponent {
     // the element's natural position(using translateY(-50%)). While in the
     // horizontal direction the fob's center is actually rendered over the left
     // of the element's natural position (using translateX(-50%)).
+    const currentFob = this.getCurrentFob()?.nativeElement;
+    if (!currentFob) {
+      return 0;
+    }
+    let fobTopPosition = currentFob.getBoundingClientRect().top;
+    let fobLeftPosition = currentFob.getBoundingClientRect().left;
+
     if (this.axisDirection === AxisDirection.VERTICAL) {
       return (
-        (this.currentDraggingFob !== Fob.END
-          ? this.startFobWrapper.nativeElement.getBoundingClientRect().top
-          : this.endFobWrapper.nativeElement.getBoundingClientRect().top) -
-        this.root.nativeElement.getBoundingClientRect().top
-      );
-    } else {
-      return (
-        (this.currentDraggingFob !== Fob.END
-          ? this.startFobWrapper.nativeElement.getBoundingClientRect().left
-          : this.endFobWrapper.nativeElement.getBoundingClientRect().left) -
-        this.root.nativeElement.getBoundingClientRect().left
+        fobTopPosition - this.root.nativeElement.getBoundingClientRect().top
       );
     }
+    return (
+      fobLeftPosition - this.root.nativeElement.getBoundingClientRect().left
+    );
   }
 
-  getDraggingFobStep(): number {
-    return this.currentDraggingFob !== Fob.END
-      ? this.timeSelection!.start.step
-      : this.timeSelection!.end!.step;
+  getCurrentFob(): ElementRef<CardFobComponent & HTMLElement> | null {
```

Comment:
I like this function. Making things clean.
QQ: after lunching this feature, can we remove null from the return type? If so, please add TODO

Reply:
> prospective fob feature is not enabled

do you mean (1)feature flag not enabled or (2)after launch we want to have a checkbox to disable it?

If (1), without prospective fob feature enabled there shouldn't be prospective-fob-area element at all. This method will only trigged on fob dragging then. Let me know if you tested it and it did broken. 

And yes, please add the note anywhere (in internal list is fine too) I want to get ride of null type eventually.
In addition, I think that's a lot of dependency. to remember what method could be triggered on what condition. Not sure how you feel but I don't like it.
e.g. the train currently is 
Drag action --(checking Fob.NONE) --> mouseMove() --> getNewStep() --> isMovingLower --> getCurrentStep
so since we've checked Fob.NONE and I _think_ this method should not try to locate prospectiveFobWrapper
For future hover action, I _think_ we should not even execute this method when no perspective fob visible.

Reply:
When the feature is disabled the prospective fob will not exist. This will result in `this.prospectiveFobWrapper` being `null`. Because this function gets called when the mouse is over the chart, we will encounter null pointer errors without the null check.